lld/ELF/Driver.cpp

```diff
@@ -2518,9 +2521,9 @@ static void combineVersionedSymbol(Symbol &sym,
     sym.symbolKind = Symbol::PlaceholderKind;
     sym.isUsedInRegularObj = false;
   } else if (auto *sym1 = dyn_cast<Defined>(&sym)) {
-    if (sym2->versionId > VER_NDX_GLOBAL
-            ? config->versionDefinitions[sym2->versionId].name == suffix1 + 1
-            : sym1->section == sym2->section && sym1->value == sym2->value) {
+    if (sym2->versionId == VER_NDX_GLOBAL || sym2->versionId == nonExported
```

Comment:
I think this would be easier to read with parens, the operator precedence is not necessarily obvious.

Reply:
I've seen some people use `(a && b) ? c : d`, but this isn't a majority.

Reply:
In that case we should keep to the more common style. I think I just found it slightly harder to parse since I initially viewed this diff on mobile.